js/web/lib/wasm/jsep/webgpu/ops/cumsum.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {DataType} from '../../../wasm-common';
+import {TensorView} from '../../tensor-view';
+import {ShapeUtil} from '../../util';
+import {AttributeWithCacheKey, createAttributeWithCacheKey} from '../attribute-with-cache-key';
+import {ComputeContext, ProgramInfo, ProgramUniform} from '../types';
+
+import {createTensorShapeVariables, enableShapesUniforms, inputVariable, outputVariable, ShaderHelper} from './common';
+
+
+export interface CumSumAttributes extends AttributeWithCacheKey {
+  readonly exclusive: number;
+  readonly reverse: number;
+}
+const createCumsumProgramInfo =
+    (inputType: number, inputShape: readonly number[], axisInput: TensorView, attributes: CumSumAttributes):
+        ProgramInfo => {
+          const outputSize = ShapeUtil.size(inputShape);  // outputShape is same as inputShape.
+          const enableShapeUniforms = enableShapesUniforms(inputShape.length);
+          const shapeOrRank = enableShapeUniforms ? inputShape.length : inputShape;
+          const input = inputVariable('input', inputType, shapeOrRank);
+          const output = outputVariable('output', inputType, shapeOrRank);
+          const axisValue = axisInput.dataType === DataType.int32 ? axisInput.getInt32Array()[0] :
+                                                                    Number(axisInput.getBigInt64Array()[0]);
+          const axis = axisValue >= 0 ? axisValue : axisValue + inputShape.length;
+          const programUniforms: ProgramUniform[] = [{type: 'uint32', data: outputSize}, {type: 'int32', data: axis}];
+          if (enableShapeUniforms) {
+            const tmpProgramUniform = createTensorShapeVariables(inputShape);
+            programUniforms.push(...tmpProgramUniform);  // For input
+            programUniforms.push(...tmpProgramUniform);  // For output
+          }
+          const index = `i32(${input.indicesGet('inputIndices', 'uniforms.axis')})`;
+          const max = enableShapeUniforms ?
+              (inputShape.length === 1 ? 'i32(uniforms.input_shape)' : 'i32(uniforms.input_shape[uniforms.axis])') :
+              (inputShape.length === 1 ? 'i32(input_shape)' : 'i32(input_shape[uniforms.axis])');
+          let lowerLimit = attributes.reverse === 1 ? index : '0';
+          let upperLimit = attributes.reverse === 1 ? max : index + ' + 1';
+          if (attributes.exclusive === 1) {
+            if (attributes.reverse === 1) {
+              lowerLimit = lowerLimit + ' + 1';
+            } else {
+              upperLimit = upperLimit + ' - 1';
+            }
+          }
```

Comment:
Please move the logic of `index`, `max`, `lowerLimit` and `upperLimit` into `getShaderSource` to simplify the body of `createCumsumProgramInfo`.